File: Source/ObjectiveC/TCZoomingImageView.m
//
//  TCZoomingImageView.m
//  TCZoomingImageView
//
//  Created by Tianchi Wang on 2018/9/12.
//  Copyright © 2018 Tianchi Wang. All rights reserved.
//

#import "TCZoomingImageView.h"


@interface TCZoomingImageView()<UIScrollViewDelegate>

@property(nonatomic, strong)UIImageView *imageView;
@property(nonatomic, strong)UIScrollView *baseView;

@end


@implementation TCZoomingImageView

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self commonInit];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        self.imageView = [aDecoder decodeObjectForKey:@"imageView"];
        self.baseView = [aDecoder decodeObjectForKey:@"baseView"];
    }
    return self;
}


- (void)setUpImageView:(UIImage *)image {
    self.imageView.image = image;
}

- (void)commonInit {
    self.baseView = [[UIScrollView alloc] init];
    self.baseView.delegate = self;
    self.baseView.bounces = NO;
    self.baseView.bouncesZoom = NO;
    self.baseView.showsVerticalScrollIndicator = NO;
    self.baseView.showsHorizontalScrollIndicator = NO;
    self.baseView.maximumZoomScale = 6;
    self.baseView.translatesAutoresizingMaskIntoConstraints = NO;
    [self addSubview:self.baseView];
    
    NSLayoutConstraint *baseViewLeftLC = [self.baseView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor];
    NSLayoutConstraint *baseViewRightLC = [self.baseView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor];
    NSLayoutConstraint *baseViewBottomLC = [self.baseView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor];
    NSLayoutConstraint *baseViewTopLC = [self.baseView.topAnchor constraintEqualToAnchor:self.topAnchor];
    
    [NSLayoutConstraint activateConstraints:@[baseViewLeftLC, baseViewRightLC, baseViewBottomLC, baseViewTopLC]];
    
    self.imageView = [[UIImageView alloc] init];
    self.imageView.contentMode = UIViewContentModeScaleAspectFit;
    self.imageView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.baseView addSubview:self.imageView];
    
    NSLayoutConstraint *imageViewLeftLC = [self.imageView.leadingAnchor constraintEqualToAnchor:self.baseView.leadingAnchor];
    NSLayoutConstraint *imageViewRightLC = [self.imageView.trailingAnchor constraintEqualToAnchor:self.baseView.trailingAnchor];
    NSLayoutConstraint *imageViewBottomLC = [self.imageView.bottomAnchor constraintEqualToAnchor:self.baseView.bottomAnchor];
    NSLayoutConstraint *imageViewTopLC = [self.imageView.topAnchor constraintEqualToAnchor:self.baseView.topAnchor];
    NSLayoutConstraint *imageViewHeightLC = [self.imageView.heightAnchor constraintEqualToAnchor:self.baseView.heightAnchor];
    NSLayoutConstraint *imageViewWidthLC = [self.imageView.widthAnchor constraintEqualToAnchor:self.baseView.widthAnchor];
    
    [NSLayoutConstraint activateConstraints:@[imageViewLeftLC, imageViewRightLC, imageViewBottomLC, imageViewTopLC, imageViewHeightLC, imageViewWidthLC]];
    
    [self.baseView layoutIfNeeded];
    [self.imageView layoutIfNeeded];
}

@end


@implementation TCZoomingImageView(scrollviewDelegate)

- (UIView *)viewForZoomingInScrollView:(UIScrollView *)scrollView {
    return self.imageView;
}


@end

File: Pod/Swift/TCZoomingImageView.swift
////
////  ZoomingImageView.swift
////  TCZoomingImageView
////
////  Created by Tianchi Wang on 2018/5/5.
////  Copyright © 2018年 Tianchi Wang. All rights reserved.
////
//
//import UIKit
//
//class TCZoomingImageView: UIView {
//
//    private var baseScrolView: UIScrollView?
//    private var imageView: UIImageView?
//
//    override init(frame: CGRect) {
//        super.init(frame: frame)
//        let baseFrame = CGRect.init(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
//
//        baseScrolView = UIScrollView.init(frame: baseFrame)
//        baseScrolView?.delegate = self
//        baseScrolView?.bounces = false
//        baseScrolView?.bouncesZoom = false
//        baseScrolView?.showsVerticalScrollIndicator = false
//        baseScrolView?.showsHorizontalScrollIndicator = false
//        baseScrolView?.maximumZoomScale = 6
//
//        self.addSubview(baseScrolView!)
//        imageView = UIImageView.init(frame: baseFrame)
//        baseScrolView!.addSubview(imageView!)
//    }
//
//    required init?(coder aDecoder: NSCoder) {
//        super.init(coder: aDecoder)
//    }
//
//    override func awakeFromNib() {
//        super.awakeFromNib()
//        let baseFrame = CGRect.init(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
//
//        baseScrolView = UIScrollView.init(frame: baseFrame)
//        baseScrolView?.delegate = self
//        baseScrolView?.bounces = false
//        baseScrolView?.bouncesZoom = false
//        baseScrolView?.showsVerticalScrollIndicator = false
//        baseScrolView?.showsHorizontalScrollIndicator = false
//        baseScrolView?.maximumZoomScale = 6
//
//        self.addSubview(baseScrolView!)
//        imageView = UIImageView.init(frame: baseFrame)
//        imageView?.contentMode = .scaleAspectFit
//        baseScrolView!.addSubview(imageView!)
//    }
//
//    func setUpImageView(withImage image:UIImage) {
//        imageView?.image = image
//    }
//}
//
//extension TCZoomingImageView: UIScrollViewDelegate {
//    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
//        return imageView
//    }
//}
